staging/src/k8s.io/api/core/v1/types.go

```diff
@@ -181,6 +181,22 @@ type VolumeSource struct {
 	//
 	// +optional
 	Ephemeral *EphemeralVolumeSource `json:"ephemeral,omitempty" protobuf:"bytes,29,opt,name=ephemeral"`
+	// image represents an OCI object (a container image or artifact) pulled and mounted on the kubelet's host machine.
+	// The volume is resolved at pod startup depending on which PullPolicy value is provided:
+	//
+	// - Always: the kubelet always attempts to pull the reference. Container creation will fail If the pull fails.
+	// - Never: the kubelet never pulls the reference and only uses a local image or artifact. Container creation will fail if the reference isn't present.
```

Comment:
is there any case when the container image would be re-pulled by the kubelet and the volume image would not be re-pulled?

is there any case where the container image would not be re-pulled by the kubelet and the volume image would be re-pulled?

Reply:
> is there any case when the container image would be re-pulled by the kubelet and the volume image would not be re-pulled?

If the container image is set to a policy of `Always` and the volume image to `Never` (or `IfNotPresent` and existing on disk).

> is there any case where the container image would not be re-pulled by the kubelet and the volume image would be re-pulled?

The other way around: If the volume image is set to a policy of `Always` and the container image to `Never` (or `IfNotPresent` and existing on disk).